apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/actions.ts

```diff
@@ -23,12 +26,15 @@ export async function updateSurveyAction(survey: TSurvey): Promise<TSurvey> {
   return await updateSurvey(_survey);
 }
 
-export const deleteSurveyAction = async (surveyId: string) => {
+export const deleteSurveyAction = async (surveyId: string, environmentId: string) => {
   const session = await getServerSession(authOptions);
   if (!session) throw new AuthorizationError("Not authorized");
 
   const isAuthorized = await canUserAccessSurvey(session.user.id, surveyId);
   if (!isAuthorized) throw new AuthorizationError("Not authorized");
 
+  const { hasDeleteAccess } = await verifyUserRoleAccess(environmentId, session.user.id);
```

Comment:
we have the same action problem as mentioned below. Please derive environmentId instead of param.